### PR TITLE
Assign NPC's as vehicle crew members

### PR DESCRIPTION
**Pull request by mugling**
Initial support for assigning crew members to certain vehicle parts via the vehicle interaction menu

A crew member is an NPC who must be:
- alive, specifically `!npc::is_dead_state()`
- allied, specifically `npc::is_friend()`

Vehicle parts that can be assigned crew must be:
- attached, specifically `!is_broken()`
- either `is_seat()` or `is_turret()`

If you assign a crew member a seat they will prefer this over others when boarding a vehicle and other NPc's will be excluded from that selecting that seat. @Coolthulhu's default seat selection algorithm is preserved so NPc's that don't have a specific seat will select from the pool of unassigned seats.

The next PR will handle assigning NPC's as gunners for turrets (which requires some system of target designation - suggestions welcomed).

### CODE REVIEW

File: src/vehicle.cpp [hunk 1]
```diff
@@ -6197,6 +6220,42 @@ int vehicle_part::wheel_width() const
...
+    if( is_broken() || ( !is_seat() && !is_turret() ) ) {
+        return false;
+    }
+    crew_id = who.getID();
+    return true;
+}
+
+void vehicle_part::unset_crew()
+{
+    crew_id = 0;
```
Comment (Coolthulhu) [inline src/vehicle.cpp]: Isn't 0 a valid character id?

**Comment by mugling** [comment]
Ready

File: src/vehicle.cpp (hunk 1 above)
Comment (mugling) [inline src/vehicle.cpp]: See 2f992a8

**Comment by mugling** [comment]
We could probably do with running boards for NPC's to stand on (separate PR)

File: src/vehicle.cpp [hunk 2]
```diff
@@ -6197,6 +6220,42 @@ int vehicle_part::wheel_width() const
     return base.is_wheel() ? base.type->wheel->width : 0;
 }
 
+npc * vehicle_part::crew()
```
Comment (Coolthulhu) [inline src/vehicle.cpp]: Why not `const`?

Comment (mugling) [inline src/vehicle.cpp]: No good reason actually, I'll change that

Comment (mugling) [inline src/vehicle.cpp]: Fixed in 8ec971a

**Comment by DangerNoodle** [comment]
As I intended to post before a connection outage interrupted me, I do like the potential this has, and have a few possible suggestions related to the future implementation of NPCs using turrets.

One is the obvious, it might warrant adding an option for NPCs to be commanded whether or not to use the gun installed at their seat. The player might want to control whether a turret is made use of.

Another is the issue of how the NPC should handle reloading a turret. If there are any plans to implement such a feature, this leads to the question of whether the player should have control over whether or not they reload the turret they control.

**Comment by Coolthulhu** [comment]
Minor:
* In crew option, when selecting a part, keybinds on the left don't work (there is abcd etc. listing, but pressing 'a' does nothing)
* No way to unassign crew
* Can't assign without letting go of the controls
* The "no NPCs to assign" message is immediately overwritten and can't be seen

**Comment by mugling** [comment]
> The "no NPCs to assign" message is immediately overwritten and can't be seen

This appears to have become a more general problem with the vehicle UI (for example low morale now stops vehicle part installation without any obvious explanation)

**Pull request merged**